### ISSUE TITLE
Server-client config mismatch causes de-syncs

### ISSUE THREAD

**Issue by lashloch** [Insane-96/IguanaTweaksReborn#24]
Some weird things happen when the block hardness multiplier variable in the server's config is different from the client's.

For instance, my server config is set to multiplier=16.0. If my client config is set to multiplier=8.0 and I attempt to break a block, the block's "breaking" animation will play, the block will pop, but then the block returns to its original undamaged appearance. Then, a second or two later, without touching the block further, the block breaks and drops an item. Kind of like server lag, but not (testing on localhost, so there should be no lag).

Basically, blocks appear to break according to the client's multiplier, not the server's. This would be fine, if all it did was play the breaking animation several times. But you can stop hitting the block after it pops the first time and simply wait for the block to drop. Doesn't seem like the intended effect.

Now, I haven't tested it to see if I can break blocks faster than the server should allow by manipulating my client's config, but I don't really want to take the chance on my public server.

Let me know if there's something I've missed configuration-wise. Thanks!

**Comment by Insane-96** [comment]
> Let me know if there's something I've missed configuration-wise. Thanks!

Sadly you didn't miss anything, that's a pretty nifty bug, I really don't know how to fix it. I thought that server config would automatically override client one, but doesn't happen

**Comment by lashloch** [comment]
Seems tricky. I did some googling and found a forum thread that might help: [](http://www.minecraftforge.net/forum/topic/56315-sync-config-between-client-and-server/)

**Comment by Insane-96** [comment]
So. I need to refactor the config part of the mod to fix this. Well, for now just avoid having different values between client and server 😞 
This will take a bit

**Comment by Insane-96** [comment]
It took just 7-8 months. But it's coming
